### PR TITLE
agent/caching: refactor ContextInfo

### PR DESCRIPTION
**Pull request by calvn**

### CODE REVIEW

File: command/agent/cache/cachememdb/index.go [hunk 1]
```diff
@@ -95,3 +89,20 @@ func validIndexName(indexName string) bool {
 	}
 	return true
 }
+
+type ContextInfo struct {
```
Comment (vishalnayak) [inline command/agent/cache/cachememdb/index.go]: It makes more sense to have this in lease cache than in cachememdb. Don't you think?

Comment (calvn) [reply command/agent/cache/cachememdb/index.go]: cachememdb still needs this and package cache imports it, so it would cause a dep cycle.

Comment (vishalnayak) [reply command/agent/cache/cachememdb/index.go]: I see. 👍

File: command/agent/cache/lease_cache.go [hunk 2]
```diff
@@ -108,7 +101,7 @@ func NewLeaseCache(conf *LeaseCacheConfig) (*LeaseCache, error) {
 
 	// Create a base context for the lease cache layer
 	baseCtx, baseCancelFunc := context.WithCancel(conf.BaseContext)
-	baseCtxInfo := &ContextInfo{
+	baseCtxInfo := &cachememdb.ContextInfo{
```
Comment (vishalnayak) [inline command/agent/cache/lease_cache.go]: Can we use `cachememdb.NewContextInfo` here by passing in `conf.BaseContext` as parameter?

Comment (calvn) [reply command/agent/cache/lease_cache.go]: Fixed in d120680.

**Pull request merged**